core/test/libraries/Positions.t.sol

```diff
@@ -19,6 +19,57 @@ contract PositionsTest is Test {
         positions[5] = 678;
     }
 
+    function test_memoryZip(int24 xl, int24 xu, int24 yl, int24 yu, int24 zl, int24 zu) public {
+        int24[6] memory arr = [xl, xu, yl, yu, zl, zu];
+        vm.expectSafeMemory(0x00, 0x60);
+        zip(arr);
+    }
+
+    function test_memoryExtract(int24 xl, int24 xu, int24 yl, int24 yu, int24 zl, int24 zu) public {
+        uint64 count = 0;
+        if (xl != xu) count += 1;
+        if (yl != yu) count += 1;
+        if (zl != zu) count += 1;
+        uint256 zipped = zip([xl, xu, yl, yu, zl, zu]);
+
+        uint64 ptr;
+        assembly ("memory-safe") {
+            ptr := mload(0x40)
+        }
+
+        vm.expectSafeMemory(ptr, ptr + 32 * (1 + 2 * count));
```

Comment:
Took some time to wrap my head around this, but this makes sense. Start at the free memory pointer and then move in 32 byte increments depending on how many elements there are in our compressed data format (which is just one number). 👍